src/backend/logger/logger.ts

```diff
@@ -504,6 +538,28 @@ export async function initGamePlayLog(gameInfo: GameInfo) {
   return logsWriters[`${gameInfo.app_name}-lastPlay`].initLog()
 }
 
+export async function appendWinetricksGamePlayLog(gameInfo: GameInfo) {
+  const logWriter = logsWriters[`${gameInfo.app_name}-lastPlay`]
+  if (logWriter) {
+    // append a promise to the queue
+    logWriter.logMessage(
+      new Promise((resolve, reject) => {
+        Winetricks.listInstalled(gameInfo.runner, gameInfo.app_name)
+          .then((installedPackages) => {
+            const packagesString = installedPackages
+              ? installedPackages.join(', ')
+              : 'none'
+
+            resolve(`Winetricks packages: ${packagesString}\n\n`)
+          })
+          .catch((error) => {
+            reject(error)
+          })
+      })
+    )
```

Comment:
This feels strange to me. Why not await the promise in here and then just log the packages, avoiding all the "You can put a Promise<string | string[]> into the queue" stuff
```suggestion
  const installedPackages = await Winetricks.listInstalled(gameInfo.runner, gameInfo.app_name)
  const packagesString = installedPackages
    ? installedPackages.join(', ')
    : 'none'
  logWriter.logMessage(`Winetricks packages: ${packagesString}\n\n`)
```

Reply:
The problem of waiting for the promise here is that it takes some time to wait and during that time other messages are appended to the log (the game starts and sends output), so this ends up mixed with the game output instead of appended to the logs before the game launches

with this approach, the promise gets enqueued before the game starts, so the winetricks info is consistently logged before the game output cause it's before those in the messages queue

Reply:
I see, hadn't considered that. This is quite clever!